### PR TITLE
feat: Decrypt but ignore messages from legal hold clients

### PR DESCRIPTION
**Pull request by bennyn**
It looks like `_distributeEvent` is a "fire and forget" method which is why I am not awaiting it in `_handleEventDistribution` of `EventRepository.js`. Or have I overlooked something?

### CODE REVIEW

File: src/script/event/EventRepository.js [hunk 1]
```diff
@@ -636,6 +644,7 @@ export class EventRepository {
         }, Promise.resolve(mappedEvent));
       })
       .then(mappedEvent => {
+        mappedEvent.from_client = event.data && event.data.sender ? event.data.sender : '';
```
Comment (ffflorian) [inline src/script/event/EventRepository.js]:
```suggestion
        mappedEvent.from_client = event.data && event.data.sender || '';
```

File: src/script/event/EventRepository.js [hunk 2]
```diff
@@ -565,14 +565,22 @@ export class EventRepository {
    * @param {EventRepository.SOURCE} source - Source of notification
    * @returns {undefined} No return value
```
Comment (ffflorian) [inline src/script/event/EventRepository.js]:
```suggestion
   * @returns {Promise<void>} Empty promise
```

**Comment by ffflorian** [comment]
Even though `_distributeEvent` is a "fire-and-forget" method, it should still be awaited.

**Comment by bennycode** [comment]
@ffflorian It depends. Useless awaits can also make the app slower but to be on a safe side I put an `await` now. :)

**Comment by ffflorian** [comment]
> Useless awaits can also make the app slower

`await` should be non-blocking, do you have documentation on it slowing apps down?

**Comment by bennycode** [comment]
> await should be non-blocking, do you have documentation on it slowing apps down?

Yes, I do: https://medium.com/front-end-weekly/async-await-is-not-about-making-asynchronous-code-synchronous-ba5937a0c11e

Check the article's sample with:

```
const data1 = await downloadFromService1();
const data2 = await downloadFromService2();
const data3 = await downloadFromService3();
```

**Comment by bennycode** [comment]
Requirements have changed. Functionality is not needed at this point in time.

**Pull request closed without merging**